### DISCUSSION TITLE
How to use plugins?

### DISCUSSION THREAD

**Discussion by Okiharaherbst** [FranckFreiburger/vue3-sfc-loader#56] [Q&A] [answered]
Thanks for this great library, looks promising for a small project I'd like to get off the ground.

I'm just a bit unsure as to how plugins should be used when creating the Vue app instance.
The examples from the Vue documentation show:

`import i18nPlugin from './plugins/i18n'`

But how should I use this without webpack?

**Comment by FranckFreiburger** [comment, marked as answer]
Have a look at the following example : https://github.com/FranckFreiburger/vue3-sfc-loader/blob/main/docs/examples.md#use-sfc-custom-blocks-for-i18n

Note that you can also load locales from JSON files

**Reply by Okiharaherbst** [reply]
Thanks, the focus though was on how to load plugins and not specifically i18n centric.

If I'm following, you would just define the plugin and load it through a script tag?

**Reply by FranckFreiburger** [reply]
**1/**
If your plugin is reachable from the current URL (relative or absolute) just use `import `or `require()`.
(vue3-sfc-loader is able to load esm and cjs module).
eg.
```
import myPlugin from '../plugins/myPlugin/index.mjs'
```

**2/**
If the plugin is loaded through a `<script>` tag, you have to store the plugin's module in `options.moduleCache`.
Note that this is what is currently done with vue :
```
const options = {
      moduleCache: {
        vue: Vue
      },
      ...
```
Subsequent `import Vue from 'vue'` will return the Vue module.

**3/**
If your plugin is hosted elsewhere (on a different origin), you have to define your own `option.pathResolve`. See https://github.com/FranckFreiburger/vue3-sfc-loader/blob/main/docs/examples.md#use-remote-components

**Reply by FranckFreiburger** [reply]
see https://github.com/FranckFreiburger/vue3-sfc-loader/blob/main/docs/faq.md#how-to-use-3rd-party-plugins-or-components

**Reply by Okiharaherbst** [reply]
Got it, thanks!
